src/zjs_ocf_server.c

```diff
@@ -461,47 +476,86 @@ static jerry_value_t ocf_register(const jerry_value_t function_val,
     }
     jerry_release_value(secure_val);
 
+    resource_list_t* new = zjs_malloc(sizeof(resource_list_t));
+    if (!new) {
+        jerry_release_value(res_type_array);
+        REJECT(promise, "InternalError", "Could not allocate resource list", h);
+        return promise;
+    }
     resource = new_server_resource(resource_path);
-
-    if (zjs_ocf_start() < 0) {
-        REJECT(promise, "InternalError", "OCF failed to start", h);
+    if (!resource) {
+        jerry_release_value(res_type_array);
+        REJECT(promise, "InternalError", "Could not allocate resource", h);
         return promise;
     }
+    new->resource = resource;
+    new->next = res_list;
+    res_list = new;
+
+    resource->flags = flags;
 
-    resource->res = oc_new_resource(resource_path, num_types, 0);
+    resource->num_types = jerry_get_array_length(res_type_array);
+    resource->resource_types = zjs_malloc(sizeof(char*) * resource->num_types);
```

Comment:
A trick that might be worth employing here (not needed in this patch) is to move resource_types and _ifaces to a data[] portion at the end of the struct. Then allocate a buffer for the struct that is sizeof(struct) + sizeof(char *) * (num_types + num_ifaces) or whatever, write those two arrays to offsets into that data, and then access them that way.

This would allow us to use one malloc instead of three which would save overhead and wasted padding I think.